### PR TITLE
Initial bindings - Table and Route services

### PR DESCRIPTION
**Pull request by mjkillough**
This adds initial bindings for the Table and Route services.

The OSRM API uses a lot of C++ templates, so it is not easy to generate Rust bindings for it. The templates confuse [`bindgen`](https://github.com/rust-lang/rust-bindgen), so we'd be left creating them by hand. This instead uses the [`libosrmc`](https://github.com/daniel-j-h/libosrmc) library, which is a simple C wrapper around the C++ API, and then builds a Rust API on top of that.

The `osrmc-sys` crate should be complete, but there are other services we can add safe wrappers to in the `osrm-rs` bindings if we'd like.

Some things I haven't done:
 - I haven't added any tests, as it isn't clear how we can best test this. If we download a random OSM map, how do we assert the routes are correct?
 - No documentation yet, but we can come back and add that later.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn it_works() {
+        let osrm = Osrm::new("./data/1.osrm").expect("uh oh");
```
Comment (me) [inline src/lib.rs]: `osrm-c` has a Makefile to download an `.osm` file and make it into `.osrm` for testing. Should we have something similar?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# osrm-rs
+
+Rust bindings for Open Source Routing Machine (OSRM).
+
+## Developing
+
+```sh
+# Install OSRM and its dependencies.
+brew install boost tbb osrm-backend
```
Comment (Antti) [inline README.md]: Wouldn't brew install dependencies automatically?

Comment (mjkillough) [reply README.md]: Good point!

File: src/lib.rs (hunk 1 above)
Comment (mjkillough) [reply src/lib.rs]: Yeah, we can do something similar. I might check the returned duration is non-zero, but not much more. 

(I wasn't aware of osrm-c. I think libosrmc might be a little bit more 'official', but it's good to know there are alternatives!)

File: README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+
+## Testing
+
+Requires: `wget`, `osrm-bckend`, `docker`
+
+```
+# First download/process the required maps:
+./prepare-test-data
```
Comment (me) [inline README.md:43]: Missing `.sh`?

**Review by me — APPROVED** [review]
LGTM!

**Pull request merged**